### ISSUE TITLE
useSortedAttributes corrupts JSX by fusing boolean shorthand attrs into neighbor names

### ISSUE THREAD

**Issue by alexrqs** [biomejs/biome#10229]
## Description

`useSortedAttributes` fuses a JSX boolean shorthand attribute (no value) with its value-bearing neighbor when sorting, producing a single invalid attribute instead of two distinct ones.

## Reproduction

**Input:**

```jsx
<button className="btn btn-primary" style={{ pointerEvents: 'none' }} disabled>
  Crear Reporte
</button>
```

**Command:**

```
biome check --write --only=assist/source/useSortedAttributes
```

**Expected output:**

```jsx
<button className="btn btn-primary" disabled style={{ pointerEvents: 'none' }}>
  Crear Reporte
</button>
```

**Actual output:**

```jsx
<button className="btn btn-primary" disabledstyle={{ pointerEvents: 'none' }}>
  Crear Reporte
</button>
```

The boolean shorthand `disabled` got concatenated into `style`'s name, producing `disabledstyle={...}` — a single invalid attribute. We hit the same pattern with `disabled` + `value` → `disabledvalue` (4 occurrences across our codebase, all flagged by `eslint react/no-unknown-property`). Silent fusions where the resulting name happens to look valid would be even worse.

## Environment

- Biome **2.4.14** (latest at time of report)
- Config extends `ultracite/biome/{core,react,next,jest}`
- Formatter disabled across all languages so the only mutation in the diff is the assist itself.

## Notes

Distinct from #10163 (which fixed nested-JSX attribute corruption). This bug occurs at single-element level when a boolean shorthand attribute is adjacent to a value-bearing at

**Comment by ematipico** [comment]
Please use our templates and fill our information
